### PR TITLE
UIOR-147 Align POL Payment and Receipt status with PO Status

### PR DESCRIPTION
**Pull request by AndreiKandratsenka**
## Purpose
it need to provide only available options for dropdowns
https://issues.folio.org/browse/UIOR-147

## Approach
created maps return available options lists depends on PO workflow status

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/components/POLine/POLineDetails/FieldReceiptStatus.js [hunk 1]
```diff
@@ -13,28 +14,53 @@ export const RECEIPT_STATUS = {
...
+                <FormattedMessage
+                  id={`ui-orders.receipt_status.${key}`}
+                  key={key}
+                >
+                  {(message) => <option value={RECEIPT_STATUS[key]}>{message}</option>}
+                </FormattedMessage>
+              ))}
+            </React.Fragment>
+          )
+          : (<option value="" />)
```
Comment (aliaksei-chumakou) [inline src/components/POLine/POLineDetails/FieldReceiptStatus.js]: you have this empty option in either case, maybe it makes sense to put it above the condition?

File: src/components/POLine/POLineDetails/FieldPaymentStatus.js [hunk 2]
```diff
@@ -13,28 +14,53 @@ const PAYMENT_STATUS = {
...
-        </FormattedMessage>
-        {Object.keys(PAYMENT_STATUS).map((key) => (
-          <FormattedMessage
-            id={`ui-orders.payment_status.${key}`}
-            key={key}
-          >
-            {(message) => <option value={PAYMENT_STATUS[key]}>{message}</option>}
-          </FormattedMessage>
-        ))}
+        {PAYMENT_STATUSES_BY_ORDER_STATUS[workflowStatus]
```
Comment (aliaksei-chumakou) [inline src/components/POLine/POLineDetails/FieldPaymentStatus.js]: Basically, I think that code after `.map(` you can leave as it was. Just define right option keys array in `render`

File: src/components/POLine/POLineDetails/FieldPaymentStatus.js [hunk 3]
```diff
@@ -13,28 +14,53 @@ const PAYMENT_STATUS = {
   pending: 'Pending',
 };
 
+const PAYMENT_STATUSES_BY_ORDER_STATUS = {
+  Pending: [
```
Comment (aliaksei-chumakou) [inline src/components/POLine/POLineDetails/FieldPaymentStatus.js]: this looks like values from back-end, maybe it's better to have them as constants

File: src/components/POLine/POLineDetails/FieldReceiptStatus.js (hunk 1 above)
Comment (AndreiKandratsenka) [reply src/components/POLine/POLineDetails/FieldReceiptStatus.js]: I tried to do like that that, but it doesn't work properly. Selector component expected only <option> child.

File: src/components/POLine/POLineDetails/FieldPaymentStatus.js (hunk 3 above)
Comment (AndreiKandratsenka) [reply src/components/POLine/POLineDetails/FieldPaymentStatus.js]: Thanks, I'll do it.

File: src/components/POLine/POLineDetails/FieldPaymentStatus.js (hunk 2 above)
Comment (AndreiKandratsenka) [reply src/components/POLine/POLineDetails/FieldPaymentStatus.js]: fixed

File: src/components/POLine/POLineDetails/FieldReceiptStatus.js (hunk 1 above)
Comment (AndreiKandratsenka) [reply src/components/POLine/POLineDetails/FieldReceiptStatus.js]: thank you for explanation

File: src/components/POLine/POLineDetails/FieldReceiptStatus.js [hunk 4]
```diff
@@ -13,18 +15,33 @@ export const RECEIPT_STATUS = {
...
   render() {
+    const { workflowStatus } = this.props;
+    const statuses = RECEIPT_STATUSES_BY_ORDER_STATUS[workflowStatus] || [];
+
     return (
       <Field
         component={Select}
         label={<FormattedMessage id="ui-orders.poLine.receiptStatus" />}
         name="receiptStatus"
+        disabled={!RECEIPT_STATUSES_BY_ORDER_STATUS[workflowStatus]}
```
Comment (aliaksei-chumakou) [inline src/components/POLine/POLineDetails/FieldReceiptStatus.js]:
```suggestion
        disabled={!statuses.length}
```

File: src/components/POLine/POLineDetails/FieldPaymentStatus.js [hunk 5]
```diff
@@ -13,18 +15,33 @@ const PAYMENT_STATUS = {
...
   render() {
+    const { workflowStatus } = this.props;
+    const statuses = PAYMENT_STATUSES_BY_ORDER_STATUS[workflowStatus] || [];
+
     return (
       <Field
         component={Select}
         label={<FormattedMessage id="ui-orders.poLine.paymentStatus" />}
         name="paymentStatus"
+        disabled={!PAYMENT_STATUSES_BY_ORDER_STATUS[workflowStatus]}
```
Comment (aliaksei-chumakou) [inline src/components/POLine/POLineDetails/FieldPaymentStatus.js]:
```suggestion
        disabled={!statuses.length}
```

Comment (NikitaSedyx) [reply src/components/POLine/POLineDetails/FieldPaymentStatus.js]: +1 from me

File: src/components/POLine/POLineDetails/FieldReceiptStatus.js (hunk 4 above)
Comment (AndreiKandratsenka) [reply src/components/POLine/POLineDetails/FieldReceiptStatus.js]: yep, I've thought about it to, and decided to leave as is, but of course way can apply this suggestion

File: src/components/POLine/POLineDetails/FieldPaymentStatus.js (hunk 5 above)
Comment (AndreiKandratsenka) [reply src/components/POLine/POLineDetails/FieldPaymentStatus.js]: the same as above

File: src/common/constants/poStatuses.js [hunk 6]
```diff
@@ -0,0 +1,7 @@
+export const PO_WORKFLOW_STATUSES = {
+  pending: 'Pending',
+  open: 'Open',
+  closed: 'Closed',
```
Comment (aliaksei-chumakou) [inline src/common/constants/poStatuses.js]:
```suggestion
  closed: 'Closeddsfd dsflds fdklsfdkls dsklfs',
```

**Pull request merged**